core/src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionSupervisor.scala

```diff
@@ -91,8 +96,13 @@ class SubmissionSupervisor(executionServiceCluster: ExecutionServiceCluster,
       case _ => Restart
     }
 
-    def thresholdFunc(cause: Throwable, count: Int): Unit = {
-      logger.error(s"error monitoring submission after $count times", cause)
+    def thresholdFunc(throwable: Throwable, count: Int): Unit = throwable match {
+      case MonitoredSubmissionException(workspaceName, submissionId, cause) =>
+        // increment smaRestart counter
+        restartCounter(workspaceName, submissionId, cause) += 1
+        logger.error(s"error monitoring submission $submissionId in workspace $workspaceName after $count times", cause)
+      case _ =>
+        logger.error(s"error monitoring submission after $count times", throwable)
```

Comment:
if the actor that died is passed as a parameter to `thresholdFunc` is there a moderately-acceptable way of extracting the submission id from it?